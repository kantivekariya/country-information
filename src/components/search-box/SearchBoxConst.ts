export const SEARCH_BOX_TEST_CONSTANT = {
  searchBoxId: "searchBoxId",
  searchInputId: "searchInputId",
};
