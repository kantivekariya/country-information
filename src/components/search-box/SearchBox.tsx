import "./search-box.style.scss";
import { SEARCH_BOX_TEST_CONSTANT } from "./SearchBoxConst";

interface IProps {
  placeholder: string;
  handleChange: React.ChangeEventHandler<HTMLInputElement>;
  theme: string;
}
const SearchBox = ({ placeholder, handleChange, theme }: IProps) => (
  <div
    className="search-container"
    data-testid={SEARCH_BOX_TEST_CONSTANT.searchBoxId}
  >
    <button className="search-icon-btn">
      <i className={`fas fa-search search-icon ${theme}-mode`}></i>
    </button>
    <input
      className={`search ${theme}-mode`}
      type="search"
      placeholder={placeholder}
      onChange={handleChange}
    />
  </div>
);

export default SearchBox;
