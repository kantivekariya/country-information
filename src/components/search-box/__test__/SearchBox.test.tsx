import {
  cleanup,
  fireEvent,
  prettyDOM,
  render,
  screen,
} from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import "@testing-library/jest-dom";
import SearchBox from "../SearchBox";
import { SEARCH_BOX_TEST_CONSTANT } from "../SearchBoxConst";

afterEach(cleanup);

const searchBoxProps = {
  placeholder: "Search Country",
  handleChange: () => {},
  theme: "light",
};

describe("Search box", () => {
  it("renders search box", () => {
    render(<SearchBox {...searchBoxProps} />, {
      wrapper: MemoryRouter,
    });
    const searchElement = screen.getByTestId(
      SEARCH_BOX_TEST_CONSTANT.searchBoxId
    );
    expect(searchElement).toBeInTheDocument();
    expect(searchElement).not.toBeChecked();
  });

  it("should be matches snapshot", () => {
    const { baseElement } = render(<SearchBox {...searchBoxProps} />, {
      wrapper: MemoryRouter,
    });
    expect(baseElement).toMatchSnapshot();
  });
});
