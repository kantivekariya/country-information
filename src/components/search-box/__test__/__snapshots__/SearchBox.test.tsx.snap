// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search box should be matches snapshot 1`] = `
<body>
  <div>
    <div
      class="search-container"
      data-testid="searchBoxId"
    >
      <button
        class="search-icon-btn"
      >
        <i
          class="fas fa-search search-icon light-mode"
        />
      </button>
      <input
        class="search light-mode"
        placeholder="Search Country"
        type="search"
      />
    </div>
  </div>
</body>
`;
